src/components/application_manager/include/application_manager/command_factory.h

```diff
@@ -54,6 +58,101 @@ class CommandFactory {
   virtual CommandSharedPtr CreateCommand(
       const commands::MessageSharedPtr& message,
       commands::Command::CommandSource source) = 0;
+  virtual bool IsAbleToProcess(
+      const int32_t,
+      const application_manager::commands::Command::CommandSource) const = 0;
+};
+
+class ICommandCreator {
+ public:
+  virtual ~ICommandCreator() {}
+  virtual bool isAble() const = 0;
+  virtual CommandSharedPtr create(
+      const commands::MessageSharedPtr& message) const = 0;
+};
+
+template <typename CommandType>
+class CommandCreator : public ICommandCreator {
+ public:
+  CommandCreator(ApplicationManager& application_manager,
+                 RPCService& rpc_service,
+                 HMICapabilities& hmi_capabilities,
+                 PolicyHandlerInterface& policy_handler)
+      : application_manager_(application_manager)
+      , rpc_service_(rpc_service)
+      , hmi_capabilities_(hmi_capabilities)
+      , policy_handler_(policy_handler)
+      , able_(true) {}
+
+ private:
+  bool isAble() const override;
```

Comment:
@BSolonenko I think implementation of this function can be moved to ICommandCreator, with variable,